src/core/providers/DappProvider/helpers/signErrors/handleSignError.ts

```diff
@@ -3,39 +3,46 @@ import {
   CANCEL_TRANSACTION_TOAST_ID,
   ERROR_SIGNING_TOAST_ID
 } from 'constants/transactions.constants';
+import { getLedgerErrorCodes } from 'core/providers/strategies/LedgerProviderStrategy/helpers/getLedgerErrorCodes';
 import { createCustomToast } from 'store/actions';
 import { SigningErrorsEnum, SigningWarningsEnum } from 'types/enums.types';
 
-export function handleSignError(error?: unknown) {
+const states = {
+  error: {
+    title: SigningErrorsEnum.errorSigning.toString(),
+    iconClassName: 'danger',
+    toastId: `${ERROR_SIGNING_TOAST_ID}-${Date.now()}`
+  },
+  warning: {
+    title: SigningWarningsEnum.cancelled.toString(),
+    iconClassName: 'warning',
+    toastId: `${CANCEL_TRANSACTION_TOAST_ID}-${Date.now()}`
+  }
+};
+
+export function handleSignError(
+  error: unknown,
+  type: 'error' | 'warning' = 'error'
```

Comment:
Why not an object instead of two arguments? Also, is there no way to avoid using `unknown`?